### DISCUSSION TITLE
Plausible dashboard loads endlessly

### DISCUSSION THREAD

**Discussion by nourkagha** [plausible/analytics#2881] [Self-Hosted Support]
Hello,

I'm unable to view anything on my dashboard as each panel gets stuck on loading endlessly. Things have been working fine for nearly a month and this started happening all of a sudden today.

Here are the logs for `plausible-plausible-1`:

```
00:36:07.056 [error] Task #PID<0.3216.0> started from #PID<0.3215.0> terminating
** (FunctionClauseError) no function clause matching in Clickhousex.Codec.Values.to_iodata/4
    (clickhousex 0.4.0) lib/clickhousex/codec/values.ex:120: Clickhousex.Codec.Values.to_iodata({:error, :invalid_date}, 0, {:error, :invalid_date}, [])
    (clickhousex 0.4.0) lib/clickhousex/codec/values.ex:104: Clickhousex.Codec.Values.encode_param/2
    (clickhousex 0.4.0) lib/clickhousex/codec/values.ex:39: Clickhousex.Codec.Values.weave/4
    (clickhousex 0.4.0) lib/clickhousex/query.ex:62: DBConnection.Query.Clickhousex.Query.encode/3
    (db_connection 2.4.2) lib/db_connection.ex:1255: DBConnection.encode/5
    (db_connection 2.4.2) lib/db_connection.ex:1355: DBConnection.run_prepare_execute/5
    (db_connection 2.4.2) lib/db_connection.ex:1459: DBConnection.run/6
    (db_connection 2.4.2) lib/db_connection.ex:595: DBConnection.parsed_prepare_execute/5
Function: #Function<7.114517855/0 in Plausible.Stats.Timeseries.timeseries/3>
    Args: []
00:36:07.058 [error] Ranch protocol #PID<0.3215.0> of listener PlausibleWeb.Endpoint.HTTP (connection #PID<0.3214.0>, stream id 1) terminated
an exception was raised:
    ** (FunctionClauseError) no function clause matching in Clickhousex.Codec.Values.to_iodata/4
        (clickhousex 0.4.0) lib/clickhousex/codec/values.ex:120: Clickhousex.Codec.Values.to_iodata({:error, :invalid_date}, 0, {:error, :invalid_date}, [])
        (clickhousex 0.4.0) lib/clickhousex/codec/values.ex:104: Clickhousex.Codec.Values.encode_param/2
        (clickhousex 0.4.0) lib/clickhousex/codec/values.ex:39: Clickhousex.Codec.Values.weave/4
        (clickhousex 0.4.0) lib/clickhousex/query.ex:62: DBConnection.Query.Clickhousex.Query.encode/3
        (db_connection 2.4.2) lib/db_connection.ex:1255: DBConnection.encode/5
        (db_connection 2.4.2) lib/db_connection.ex:1355: DBConnection.run_prepare_execute/5
        (db_connection 2.4.2) lib/db_connection.ex:1459: DBConnection.run/6
        (db_connection 2.4.2) lib/db_connection.ex:595: DBConnection.parsed_prepare_execute/5
```

This is what the output of the console on Chrome shows:

![Screenshot from 2023-04-27 02-48-29](https://user-images.githubusercontent.com/50773468/234732085-e1d93c9b-e3ba-43d0-ad29-3ef73cb9adcd.png)

Also, my server had been set up exactly this way on a droplet: https://www.digitalocean.com/community/tutorials/how-to-install-plausible-analytics-on-ubuntu-22-04

**Comment by ruslandoga** [comment]
@nourkagha 👋

Can you please show the HTTP request the browser makes?

**Reply by ruslandoga** [reply]
It seems like one of the dependencies fails to parse a date (timex) and then Plausible "blindly" uses it in a database query and the error tuple fails to be encoded in query params which leads to the exception you see in the logs.

**Reply by ruslandoga** [reply]
Timex is being tricky here as it doesn't wrap successful results in `{:ok, result}` tuples, but it does so for errors.

https://hexdocs.pm/timex/Timex.html#diff/3

```elixir
@spec diff(
  Time.t() | Timex.Comparable.comparable(),
  Time.t() | Timex.Comparable.comparable(),
  Timex.Comparable.granularity()
) :: Timex.Duration.t() | integer() | {:error, term()}
```

**Reply by nourkagha** [reply]
Thank you for looking into it!

Here it is via Firefox:

![Screenshot from 2023-04-27 15-59-43](https://user-images.githubusercontent.com/50773468/234885596-f1061fac-ab8d-46f6-987a-67956953eefa.png)

If needed, you can get to my dashboard here: https://plausible.agha.dev/agha.dev

**Reply by nourkagha** [reply]
Huh.. so this is very interesting. I tried to access my dashboard today and unlike before now the whole thing was down and the dashboard wouldn't open at all with a 500 error.

I SSHed into my server and checked the logs and saw something new:

```
00:18:36.645 [error] #PID<0.4626.0> running PlausibleWeb.Endpoint (connection #PID<0.4625.0>, stream id 1) terminated
Server: localhost:8000 (http)
Request: GET /agha.dev
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Timex.Timezone.total_offset/1
        (timex 3.7.8) lib/timezone/timezone.ex:469: Timex.Timezone.total_offset({:error, {:could_not_resolve_timezone, "Africa/Cairo", 63849860316, :wall}})
        (plausible 0.0.1) lib/plausible_web/templates/stats/stats.html.eex:19: PlausibleWeb.StatsView."stats.html"/1
        (phoenix_view 1.1.2) lib/phoenix/view.ex:298: Phoenix.View.render_within/3
        (phoenix_view 1.1.2) lib/phoenix/view.ex:475: Phoenix.View.render_to_iodata/3
        (phoenix 1.6.6) lib/phoenix/controller.ex:772: Phoenix.Controller.render_and_send/4
        (plausible 0.0.1) lib/plausible_web/controllers/stats_controller.ex:1: PlausibleWeb.StatsController.action/2
        (plausible 0.0.1) lib/plausible_web/controllers/stats_controller.ex:1: PlausibleWeb.StatsController.phoenix_controller_pipeline/2
        (phoenix 1.6.6) lib/phoenix/router.ex:355: Phoenix.Router.__call__/2
```

I'm currently based in Cairo and the clocks went forward tonight. I went ahead and tried changing to a different timezone (Europe/Berlin) from my Plausible site settings and.. that worked. Everything is working flawlessly again, but I'm confused what could have caused the errors, especially before today where nothing had changed. Perhaps [this](https://status.digitalocean.com/incidents/f1ksdtz0gb7s) was the cause of the issue yesterday? When I had checked yesterday, [this](https://status.digitalocean.com/incidents/5k1yppbhpsp8) was the only ongoing issue at the time and it didn't seem that it would affect Plausible.

As for the clock change a few hours ago, maybe that might have affected things tonight. Although I had another site on Cairo timezone that was still working fine. Odd, but I'm glad it's all finally resolved.
